benchmarks/RequestResponseLatency.cpp

```diff
@@ -0,0 +1,228 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <benchmark/benchmark.h>
+#include <thread>
+
+#include <folly/io/async/ScopedEventBaseThread.h>
+#include <iostream>
+#include <experimental/rsocket/transports/TcpConnectionAcceptor.h>
+#include <src/NullRequestHandler.h>
+#include <src/SubscriptionBase.h>
+#include "rsocket/RSocket.h"
+#include "rsocket/transports/TcpConnectionFactory.h"
+
+using namespace ::reactivesocket;
+using namespace ::folly;
+using namespace ::rsocket;
+
+#define MESSAGE_LENGTH (32)
+
+DEFINE_string(host, "localhost", "host to connect to");
+DEFINE_int32(port, 9898, "host:port to connect to");
+
+
+class BM_Subscription : public SubscriptionBase {
+public:
+    explicit BM_Subscription(
+        std::shared_ptr<Subscriber<Payload>> subscriber,
+        std::string name,
+        size_t length)
+        : ExecutorBase(defaultExecutor()),
+        subscriber_(std::move(subscriber)),
+        name_(std::move(name)),
+        data_(length, 'a'),
+        cancelled_(false)
+    {
+    }
+
+private:
+    void requestImpl(size_t n) noexcept override
+    {
+        LOG(INFO) << "requested=" << n;
+
+        if (cancelled_) {
+            LOG(INFO) << "emission stopped by cancellation";
+            return;
+        }
+
+        subscriber_->onNext(Payload(data_));
+        subscriber_->onComplete();
+    }
+
+    void cancelImpl() noexcept override
+    {
+        LOG(INFO) << "cancellation received";
+        cancelled_ = true;
+    }
+
+    std::shared_ptr<Subscriber<Payload>> subscriber_;
+    std::string name_;
+    std::string data_;
+    std::atomic_bool cancelled_;
+};
+
+class BM_RequestHandler : public DefaultRequestHandler
+{
+public:
+    void handleRequestResponse(
+        Payload request, StreamId streamId, const std::shared_ptr<Subscriber<Payload>> &response) noexcept override
+    {
+        LOG(INFO) << "BM_RequestHandler.handleRequestResponse " << request;
+
+        const char* p = reinterpret_cast<const char*>(request.data->data());
+        auto requestString = std::string(p, request.data->length());
```

Comment:
Could yes. This is the critical path for that benchmark, though. So, it should be simple and quick. Not sure which call is going to add noise to the measurement. And which one could be optimized best by the compiler. So, I went with one I know can normally be very efficient.